tests/unit/backend/workflow/test_engine.py

```diff
@@ -0,0 +1,113 @@
+# package imports
```

Comment:
I think it is better to
- expose the embedded class as public one
- write standard unit test for each class method, mock out details if needed
- the testing for wrapper function can be a simple verification on invoking the correct class